source/js/openstreetmap/addMarkers.ts

```diff
@@ -0,0 +1,144 @@
+
+        if (!sidebar) return;
+        const placeElements = sidebar.querySelectorAll('[data-js-map-location]');
+        
+        placeElements.forEach(element => {
+            locations.push(getMarkerDataFromElement(element as HTMLElement));
+        });
+        
+        return locations;
+    }
+
+    private addMarkersToMap() {
+        const locations = this.getAllLocations() ?? [];
+        locations.forEach((location, index) => {
+            if (location?.lat && location?.lng) {
+                let customIcon: Icon | undefined = undefined;
+                if (location?.icon) {
+                    customIcon = location.icon;
+                } 
+
+                let marker = L.marker([location.lat, location.lng], {
+                    icon: this.createMarker(customIcon),
+                    // url: { lat: location.lat, lng: location.lng },
+                    // id: `${'marker' + index}`,
+                });
+                if (location.tooltip) {
+                    marker.bindPopup(this.createTooltip(location.tooltip), { maxWidth: 300 });
+                }
+                marker.on('click', (e) => {
+                    let latlng = e.latlng
+                        ? e.latlng
+                        : e.sourceTarget?._latlng
+                            ? e.sourceTarget?._latlng
+                            : false;
+                    let zoomLevel = this.map?.getZoom() ?? 16;
+                    if (latlng) {
+                        if (zoomLevel >= 16) {
+                            this.map?.setView(latlng);
+                        } else {
+                            this.map?.setView(latlng, 16);
+                        }
+                    }
+                    pushCoordinatesToBrowserHistory({ lat: location.lat, lng: location.lng });
+                });
+                this.markers?.addLayer(marker);
+                if (location.element) {
+                    this.markerElementPairs?.push({marker: marker, element: location.element});
+                }
+            }
+        });
+        this.markers?.addTo(this.map as LeafletMap);
+    }
+
+    private createMarker(customIcon: Icon | undefined) {
+        let template = this.container?.querySelector('.c-openstreetmap__pin-icon');
+        let html = template?.innerHTML;
+        let icon = customIcon?.icon ? customIcon.icon : 'location_on';
+        let color = customIcon?.backgroundColor
+            ? customIcon.backgroundColor
+            : this.getPrimaryColor();
+        
+            if (!html) return;
+        html = html
+            .replace('{icon-name}', icon as string)
+            .replaceAll('{ICON_NAME}', icon as string)
+            .replace('{ICON_BACKGROUND_COLOR}', color as string);
+        let marker = L.divIcon({
+            className: 'c-openstreetmap__icon',
+            html: html,
+        });
+
+        return marker;
+    }
+
+    private createTooltip(tooltip: Tooltip) {
```

Comment:
```suggestion
    private getTooltipHTML(tooltip: Tooltip): string {
```

Reply:
Should this be changed?
Since it is actually creating as there is none without it. Could we name it createTooltipFromTemplate?